src/services/validatorSchema.ts

```diff
@@ -0,0 +1,17 @@
+import * as yup from 'yup';
+
+export const inventoryItemSchema = yup.object({
+    user_id: yup.string().typeError("user_id must be a string").defined("The user_id must be provided, don't forget it ! "),
+    item_id: yup.number().typeError("item_id must be a number").positive("The item_id must be a positive number.").defined("The item_id must be provided, don't forget it ! "),
+});
+
+export const userSchema = yup.object({
+    user_id: yup.number().typeError("user_id must be a number").positive("The user_id must be a positive number.").defined("The user_id must be provided, don't forget it ! ")
+});
+
+export const warehouseItemSchema = yup.object().shape({
+    name: yup.string().typeError("name must be a string").defined("The name must be provided, don't forget it ! "),
+    data: yup.object().typeError("data must be an object").defined("Some data must be provided, don't forget it ! "),
```

Comment:
data should not only be an object, it should be an object where the values are string:

```json
// this is accepted
data: {
   "key": "value"
}

// this is not:
data: {
    "key": 0
}
```

If we let the user anything else than strings, the item will be persisted in elasticsearch no problem, but the tokenization will always fail because the typing is more strict, the tokenization service expects a map of string, string and nothing else